### PR TITLE
fix(mcdu): empty atis request

### PR DESCRIPTION
**Pull request by ExampleWasTaken**
## Summary of Changes

ATIS can be requested even if no airport is specified. The returned "ATIS" will be blank. This PR fixes this behavior in that it cancels the request and displays a `NO AIRPORT SPECIFIED` message when no airport is provided.

## Additional context

Discord username (if different from GitHub): `ExampleWasTaken#0886`

## Testing instructions

- Check that no ATIS can be requested whenever the field in the top left corner is empty/displays `[ ]` and that the message `NO AIRPORT SPECIFIED` appears in the scratchpad.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_AOC_RequestsAtis.js [hunk 1]
```diff
@@ -121,6 +121,10 @@ class CDUAocRequestsAtis {
             return mcdu.getDelaySwitchPage();
         };
         mcdu.onRightInput[5] = async () => {
+            if (store["arpt1"] === "" && store["arrIcao"] === "") {
+                mcdu.addNewMessage(NXFictionalMessages.noArptSpecified);
+                return;
+            }
             store["sendStatus"] = "QUEUED";
             updateView();
             const icao = store["arpt1"] || store["arrIcao"];
```
Comment (davidwalschots) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_AOC_RequestsAtis.js:130]: **suggestion**: Move `icao` up to reduce repetition
```suggestion
            const icao = store["arpt1"] || store["arrIcao"];
            if (icao === "") {
                mcdu.addNewMessage(NXFictionalMessages.noArptSpecified);
                return;
            }
            store["sendStatus"] = "QUEUED";
            updateView();
```

**Review by davidwalschots — APPROVED** [review]
I have one suggestion, but will approve with or without it being applied.

**Review by davidwalschots — APPROVED** [review]

**Comment by BritishAvgeek** [comment]
Category: QA Tester
29/04/2021

In leiu of a full report, I tested this today and didn't get anything as described - I tried to get some blank ones in but it didn't provide the warning as described but instead just remained blank for 2-3 seconds, then the destination airport got bumped back into the field. Either I was doing something wrong or it wasn't working as expected. Tried before inserting any flight plans prior to stream and got the same behaviour

**Comment by ExampleWasTaken** [comment]
> 
> 
> Category: QA Tester
> 29/04/2021
> 
> In leiu of a full report, I tested this today and didn't get anything as described - I tried to get some blank ones in but it didn't provide the warning as described but instead just remained blank for 2-3 seconds, then the destination airport got bumped back into the field. Either I was doing something wrong or it wasn't working as expected. Tried before inserting any flight plans prior to stream and got the same behaviour

That is very interesting. I'll take a look at it.

**Comment by ExampleWasTaken** [comment]
@BritishAvgeek could you describe the steps you've taken, please?

**Comment by BritishAvgeek** [comment]
> @BritishAvgeek could you describe the steps you've taken, please?

1. Spawned C&D EGKK
2. Elec etc on, aligned ADIRS, went to MCDU and requested ATIS info via a blank airport... nothing happened
3. Imported simbrief flight plan to LFMN, returned to request ATIS page, I was able to delete LFMN to get the [ ] but when I requested the ATIS it did nothing, then bumped LFMN back into the box

**Comment by ExampleWasTaken** [comment]
> 
> 
> > @BritishAvgeek could you describe the steps you've taken, please?
> 
>     1. Spawned C&D EGKK
> 
>     2. Elec etc on, aligned ADIRS, went to MCDU and requested ATIS info via a blank airport... nothing happened
> 
>     3. Imported simbrief flight plan to LFMN, returned to request ATIS page, I was able to delete LFMN to get the [ ] but when I requested the ATIS it did nothing, then bumped LFMN back into the box
> 
>     4. I requested the ATIS for EGKK and LFMN and those worked as expected, tried to remove the airports to get the [ ] but again when requesting a blank one, nothing happened and it put LFMN back in

That sounds like work for me xD. Thanks for the review! Gonna have a look at what went wrong.

**Comment by ExampleWasTaken** [comment]
@BritishAvgeek I was not able to reproduce the issues you described. The only explanation I have is that the CI didn't built the artifact correctly as the build process didn't seem to have run and I updated the branch after your review. 
It's once again RTT

**Comment by marcman86** [comment]
Category: QA Tester
Name: marcman86#4907
Date of testing: 1.05.2021
Version of the sim: 1.15.8.0
PR Tested: #4629
Tier of testing: 1
Changes to observe: Empty ATIS request message

Testing technique: Tryout

Steps made to test the Pull Request:

1.) Spawned C&D at EDDP
2.) Turned on bat. and ext. power
3.) Request an empty ATIS (Image1)

Media:


Image1

Issues: -

Overall Rating: Good

Conclusions of the testing: Works

Hope this helps :-)

**Comment by FlightSimmer7700** [comment]
Category: QA Tester
Name: FlightSimmer7700#3468
Date of testing: 2/5/21
Version of the sim: 1.15.8.0
PR Tested: #4629 
Tier of testing: 1
Changes to observe:
ATIS request page

Testing technique: Loaded in and looked
Steps you've made to test the Pull Request:
1. Loaded in 
2. tried without a flight plan
3. tried with a flight plan

Flight notes (When done a full flight):

Media: Insert screenshots, recordings, or anything relevant to the review here.

Issues (If Any): None
Steps to achieve the issue

Overall Rating: Good
Conclusions of the testing: LGTM

**Comment by darkfury334** [comment]
Category: QA Trial
Name: DarkFury#4936
Date of testing: Current date 02/05/21
Version of the sim: 1.15.8.0
PR Tested: #4629
Tier of testing: 1
Changes to observe:
Empty atis request
Testing technique: tryout
Steps you've made to test the Pull Request:
1. Spawned C&D at egll, power bat / ext. power
2. Requested ATIS with no flight plan
3. Inserted  simbrief flight plan to EDDH . 
4. Returned to ATIS page , was able to delete EDDH to get [], requested atis. got company msg with no data and reinserted EDDH into box
5. Repeated step 4 with EGLL


Media: 
when f=plan loaded and airport removed
Microsoft Flight Simulator Screenshot 2021 05 02 - 20 14 18 03


Issues (If Any): 
When load simbrief flight plan and remove airport from atis page still get company msg with no data and automatically reinserts destination airport.


Overall Rating: To be improved
Conclusions of the testing: inital test with no flight plan works but when flight plan inserted issue occurs.

**Comment by ExampleWasTaken** [comment]
> 
> 
> Category: QA Trial
> Name: DarkFury#4936
> Date of testing: Current date 02/05/21
> Version of the sim: 1.15.8.0
> PR Tested: #4629
> Tier of testing: 1
> Changes to observe:
> Empty atis request
> Testing technique: tryout
> Steps you've made to test the Pull Request:
> 
>     1. Spawned C&D at egll, power bat / ext. power
> 
>     2. Requested ATIS with no flight plan
> 
>     3. Inserted  simbrief flight plan to EDDH .
> 
>     4. Returned to ATIS page , was able to delete EDDH to get [], requested atis. got company msg with no data and reinserted EDDH into box
> 
>     5. Repeated step 4 with EGLL
> 
> 
> Media:
> when f=plan loaded and airport removed
> Microsoft Flight Simulator Screenshot 2021 05 02 - 20 14 18 03
> 
> Issues (If Any):
> When load simbrief flight plan and remove airport from atis page still get company msg with no data and automatically reinserts destination airport.
> 
> Overall Rating: To be improved
> Conclusions of the testing: inital test with no flight plan works but when flight plan inserted issue occurs.

I'm currently looking into why this might happen. The PR can stay RTT just not be merged.

**Comment by darkfury334** [comment]
Retested the PR a few times this morning. Was unable to delete airport in atis when flight plan loaded. looks good to go. Worked as intended. Thanks

**Comment by DavidEGTE** [comment]
Category: QA Tester
Name: David#2177
Date of testing: 03/05/2021
Version of the sim: 1.15.8.0
PR Tested: #4629
Tier of testing: 1

Changes to observe:
check empty ATIS request

Testing technique: Try-out

Steps you've made to test the Pull Request:
1.  Spawned C&D
2. Ext. Bat + Batteries
3. Check ATIS on MCDU

Flight notes (When done a full flight):
N/A

Media: Insert screenshots, recordings, or anything relevant to the review here.
Microsoft Flight Simulator Screenshot 2021 05 03 - 09 38 00 72

Issues (If Any): Write any issues
N/A
Steps to achieve the issue:
N/A

Overall Rating: Good
Conclusions of the testing: GTG

**Comment by ElonKerman** [comment]
Category: QA Tester
Name: Elon Kerman#9253
Date of testing: 03/06/2021
Version of the sim: 1.15.8.0
PR Tested: #4629
Tier of testing: 1

Changes to observe:
check empty ATIS request

Testing technique: Try-out

Steps you've made to test the Pull Request:

Spawned C&D
Ext. Bat + Batteries
Check ATIS on MCDU
Flight notes (When done a full flight):
N/A

Media: Insert screenshots, recordings, or anything relevant to the review here.
Microsoft Flight Simulator Screenshot 2021 05 03 - 09 38 00 72

Issues (If Any): Write any issues
N/A
Steps to achieve the issue:
N/A

Overall Rating: Good
Conclusions of the testing: LGTM

**Pull request merged**